File: src/GameLogic.java
public class GameLogic {

	public static boolean checkForAWinner(String[] xoxoList){
		boolean isThereAWinnerYet = false;
		
		if (  
				(xoxoList[0].equals(xoxoList[1]) && xoxoList[0].equals(xoxoList[2])  && !xoxoList[0].equals(" ")) ||  // Top row
				(xoxoList[3].equals(xoxoList[4]) && xoxoList[3].equals(xoxoList[5])  && !xoxoList[3].equals(" ")) ||  // Middle row
				(xoxoList[6].equals(xoxoList[7]) && xoxoList[6].equals(xoxoList[8])  && !xoxoList[6].equals(" ")) ||  // Bottom row
				(xoxoList[0].equals(xoxoList[3]) && xoxoList[0].equals(xoxoList[6])  && !xoxoList[0].equals(" ")) ||  // Left column
				(xoxoList[1].equals(xoxoList[4]) && xoxoList[1].equals(xoxoList[7])  && !xoxoList[1].equals(" ")) ||  // Middle column
				(xoxoList[2].equals(xoxoList[5]) && xoxoList[2].equals(xoxoList[8])  && !xoxoList[2].equals(" ")) ||  // Right column
				(xoxoList[0].equals(xoxoList[4]) && xoxoList[0].equals(xoxoList[8])  && !xoxoList[0].equals(" ")) ||  // Diagonal 1
				(xoxoList[2].equals(xoxoList[4]) && xoxoList[2].equals(xoxoList[6])  && !xoxoList[2].equals(" "))     // Diagonal 2
				){
			
			isThereAWinnerYet = true;
			System.out.println("We have a winner!");
		}
		
		return isThereAWinnerYet;
	}

	public static boolean checkForATie(String[] xoxoList) {
		boolean isThereATieYet = true;

		for (int i = 0; i < xoxoList.length; i++) {
			if (xoxoList[i].equals(" ")) {
				isThereATieYet = false;
			}
		}
		if (isThereATieYet) {
			System.out
					.println("You win some, you lose some. But in this case, we both TIED!");
		}
		return isThereATieYet;
	}

	public static int randomNumber() {

		// Randomly set find an integer between 0 and 8
		int multiplier = 8;
		int randomNumber = (int) Math.round(Math.random() * multiplier);
		return randomNumber;
	}

	public static int goodRandomNumber(String[] xoxoList) {

		// Keep finding random number until it matches an unnoccupied spot
		int randomNumber = randomNumber();
		while (!xoxoList[randomNumber].equals(" ")) {
			randomNumber = randomNumber();
		}
		return randomNumber;
	}
}

File: TicTacToe/GameLogic.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TicTacToe
{
	class GameLogic
	{
		private readonly string[] xoxoList;

		public GameLogic(String[] xoxoList)
		{
			this.xoxoList = xoxoList;
		}

		public bool CheckForAWinner()
		{
			if (
					(xoxoList[0] == xoxoList[1] && xoxoList[0] == xoxoList[2] && xoxoList[0] != " ") ||  // Top row
					(xoxoList[3] == xoxoList[4] && xoxoList[3] == xoxoList[5] && xoxoList[3] != " ") ||  // Middle row
					(xoxoList[6] == xoxoList[7] && xoxoList[6] == xoxoList[8] && xoxoList[6] != " ") ||  // Bottom row
					(xoxoList[0] == xoxoList[3] && xoxoList[0] == xoxoList[6] && xoxoList[0] != " ") ||  // Left column
					(xoxoList[1] == xoxoList[4] && xoxoList[1] == xoxoList[7] && xoxoList[1] != " ") ||  // Middle column
					(xoxoList[2] == xoxoList[5] && xoxoList[2] == xoxoList[8] && xoxoList[2] != " ") ||  // Right column
					(xoxoList[0] == xoxoList[4] && xoxoList[0] == xoxoList[8] && xoxoList[0] != " ") ||  // Diagonal 1
					(xoxoList[2] == xoxoList[4] && xoxoList[2] == xoxoList[6] && xoxoList[2] != " ")     // Diagonal 2
					)
			{
				Console.WriteLine("We have a winner!");
				return true;
			}
			return false;
		}

		public bool CheckIfBoardIsFull()
		{
			for (int i = 0; i < xoxoList.Length; i++)
			{
				if (xoxoList[i] == " ")
				{
					return false;
				}
			}
			return true;
		}

		public int ComputerMoves()
		{
			Random random = new Random();
			int randomNumber;
			do
			{
				randomNumber = random.Next(9);
			} while (xoxoList[randomNumber] != " ");

			return randomNumber;
		}
	}
}
